exercises/giovanni_disiena/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,102 @@
+#include "ShapeSorter.h"
+
+/** Prints shapes of given type
+ *
+ * @param vecShape: vector of unique pointers to ShapeBase objects
+ * @param type: type of shape
+ *
+ */
+void ShapeSorter::printType(vecPtr & vecShape, std::string type) {
+  unsigned int matchCounter = 0;
+  std::cout << std::string(80, '*') << "\n";
+
+  // loop through vector of pointers to ShapeBase objects
+  for (std::vector<std::unique_ptr<ShapeBase>>::const_iterator it = vecShape.begin();
```

Comment:
You can use `auto` here which would make it much easier to read. A range based for loop could also be used so this would become:
`for (const auto &it : vecShape)`